src/server/scripts/Northrend/Naxxramas/boss_heigan.cpp

```diff
@@ -66,39 +73,30 @@ class boss_heigan : public CreatureScript
 
     struct boss_heiganAI : public BossAI
     {
-        boss_heiganAI(Creature* creature) : BossAI(creature, BOSS_HEIGAN)
+        boss_heiganAI(Creature* creature) : BossAI(creature, BOSS_HEIGAN), eruptSection(0), eruptDirection(false), safetyDance(false) { }
+
+        void Reset() override
         {
-            eruptSection = 0;
-            eruptDirection = false;
-            safetyDance = false;
-            phase = PHASE_FIGHT;
+            me->SetReactState(REACT_AGGRESSIVE);
+            _Reset();
         }
 
-        uint32 eruptSection;
-        bool eruptDirection;
-        bool safetyDance;
-        Phases phase;
-
         void KilledUnit(Unit* who) override
         {
-            if (!(rand32() % 5))
-                Talk(SAY_SLAY);
-            if (who->GetTypeId() == TYPEID_PLAYER)
-                safetyDance = false;
+            KilledUnit(who->GetTypeId() == TYPEID_PLAYER);
         }
 
-        void SetData(uint32 id, uint32 data) override
+        void KilledUnit(bool isPlayer)
```

Comment:
Doesn't https://github.com/TrinityCore/TrinityCore/pull/15558/files#diff-4f67f069bf6807dbf380b3fdf513154bR240

GetHitPlayer() already check if it's a player being hit?